src/include/duckdb/common/hash_utils.hpp

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/hash_utils.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstddef>
+#include <functional>
+
+namespace duckdb {
+
+// Util for reference wrapper hashing
+template <typename Hash>
+struct RefHash : Hash {
+	RefHash() = default;
+	template <typename H>
+	RefHash(H &&h) : Hash(std::forward<H>(h)) { // NOLINT
+	}
+
+	template <typename T>
+	std::size_t operator()(std::reference_wrapper<const T> val) const {
+		return Hash::operator()(val.get());
+	}
+	template <typename T>
+	std::size_t operator()(const T &val) const {
+		return Hash::operator()(val);
+	}
+};
+
+template <typename Equal>
+struct RefEq : Equal {
+	RefEq() = default;
+	template <typename Eq>
+	RefEq(Eq &&eq) : Equal(std::forward<Eq>(eq)) { // NOLINT
+	}
+
+	template <typename T1, typename T2>
+	bool operator()(std::reference_wrapper<const T1> lhs, std::reference_wrapper<const T2> rhs) const {
+		return Equal::operator()(lhs.get(), rhs.get());
+	}
+	template <typename T1, typename T2>
+	bool operator()(const T1 &lhs, std::reference_wrapper<const T2> rhs) const {
+		return Equal::operator()(lhs, rhs.get());
+	}
+	template <typename T1, typename T2>
+	bool operator()(std::reference_wrapper<const T1> lhs, const T2 &rhs) const {
+		return Equal::operator()(lhs.get(), rhs);
+	}
+	template <typename T1, typename T2>
+	bool operator()(const T1 &lhs, const T2 &rhs) const {
+		return Equal::operator()(lhs, rhs);
+	}
+};
```

Comment:
We have `reference_map.hpp`, which (correct me if I'm wrong), implements this functionality already.

Reply:
Let me explain more on my LRU cache implementation:
- The easiest implementation is to store key twice at both unordered map and list
- But considering the address for key inside of `list` never changes (`std::list` heap allocates a node inside), we could safely use the reference wrapper for the unordered map key (it's a space optimization)
- To support lookup for `Key` with a `unordered_map<reference<Key>, Val>`, I wrote the hash utils to access the internal value if reference wrapper

As for `reference_wrapper`, I believe it's just a handy alias with key type a reference.